### ISSUE TITLE
Feedback on rust_icu_ustring

### ISSUE THREAD

**Issue by sffc** [google/rust_icu#5]
Nothing wrong here from a functionality point of view, but it raises questions about what is the best way to represent strings when interfacing with ICU.  Many ICU APIs want UTF-16 strings but also accept UTF-8 strings.  I've been toying with the idea of making a Rust version of UnicodeString that you can toggle between UTF-8 and UTF-16 at compile time.  For now, I think the safest thing is for users to keep their strings in Rust-standard UTF-8, and only do the UTF-16 conversion at the API boundary.  Even if you have to round-trip a few times between 8 and 16, I strongly suspect that this won't be a performance bottleneck.

**Comment by filmil** [comment]
Agreed.  Especially for one-off calls, that e.g. do not appear in tight loops, it is probably just fine to take `&str`.

The value of `UChar` is in being readily available for interfacing with other ICU wrapped functionality.
